python/src/iceberg/expressions/base.py

```diff
@@ -267,6 +226,68 @@ def __str__(self) -> str:
         return "false"
 
 
+class UnboundIn(BooleanExpression, UnboundPredicate):
+    """IN operation expression"""
+
+    def __init__(self, left: "UnboundReference", right: List[Literal]):  # pylint: disable=super-init-not-called
+        self._left = left
+        self._right = right
+
+    @property
+    def left(self) -> "UnboundReference":
+        return self._left
+
+    @property
+    def right(self) -> List[Literal]:
+        return self._right
+
+    def __eq__(self, other) -> bool:
+        return id(self) == id(other) or (isinstance(other, UnboundIn) and self.left == other.left and self.right == other.right)
+
+    def __invert__(self) -> Not:
+        return Not(self)
+
+    def __repr__(self) -> str:
+        return f"UnboundIn({repr(self.left)}, {repr(self.right)})"
+
+    def __str__(self) -> str:
+        return f"({self.left} in {self.right})"
+
+    def bind(self, schema: Schema, case_sensitive: bool) -> "In":
+        return In(self.left.bind(schema, case_sensitive), self.right)
```

Comment:
Bind needs to happen for both the term and the literal. The value held by the literal must be rewritten to the type of the term. That's what `literal.to(type)` is responsible for.

Reply:
call `.to` in a list comprehension now